test/react-native/features/fixtures/android-module/BugsnagModule.java

```diff
@@ -44,6 +49,38 @@ private Configuration createConfiguration(ReadableMap options) {
       Configuration config = new Configuration(options.getString("apiKey"));
       config.setEndpoints(new EndpointConfiguration(options.getString("endpoint"), options.getString("endpoint")));
       config.setAutoTrackSessions(options.getBoolean("autoTrackSessions"));
+
+      try {
+        String appVersion = null;
+        appVersion = options.getString("appVersion");
+        config.setAppVersion(appVersion);
+      } catch (NoSuchKeyException e) {
+        // ignore NoSuchKeyException
+      }
+
+      try {
+        String appType = options.getString("appType");
+        config.setAppType(appType);
+      } catch (NoSuchKeyException e) {
+        // ignore NoSuchKeyException
+      }
+
+      try {
+        String releaseStage = options.getString("releaseStage");
+        config.setReleaseStage(releaseStage);
+      } catch (NoSuchKeyException e) {
+        // ignore NoSuchKeyException
+      }
+
+      try {
+        Set<String> enabledReleaseStages = new HashSet<String>();
+        ReadableArray ar = options.getArray("enabledReleaseStages");
+        for (int i = 0; i < ar.size(); i++) enabledReleaseStages.add(ar.getString(i));
+        config.setEnabledReleaseStages(enabledReleaseStages);
+      } catch (NoSuchKeyException e) {
+        // ignore NoSuchKeyException
+      }
```

Comment:
definitely a cleaner way to do this…

Reply:
I'm generally in favour checking for existence first rather than ignoring exceptions, but the interface doesn't provide that - `ReadableMap` isn't the friendliest class in the world.

We could consider using its toHashMap() method first, although that has its own issues (e.g. casting) - or write a helper class to at least turn each block into one line.  I guess it depends how much we anticipate doing this Vs the extra effort.

Reply:
I did look at factoring this out into generic methods for each type, but it didn't really end up much cleaner

```java
String appVersion = extractString("appVersion")
if (appVersion != null) {
  configuration.setAppVersion(appVersion)
}

String releaseStage = extractString("releaseStage")
if (releaseStage != null) {
  configuration.setReleaseStage(releaseStage)
}

Set<String> enabledReleaseStages = extractStringSet("enabledReleaseStages")
if (enabledReleaseStages != null) {
  configuration.setEnabledReleaseStages(enabledReleaseStages)
}

// ....

String extractString(String k, ReadableMap opts) {
  try {
    return opts.getString(k);
  } catch (NoSuchKeyException e) {
    return null
  }
}

Set<String> extractStringSet(String k, ReadableMap opts) {
  //...
}
```

Reply:
Yeah, just leave it as it is, I think we're in danger of over-thinking it.